src/components/PeopleTable/PeopleTable.tsx

```diff
@@ -0,0 +1,148 @@
+import classNames from 'classnames';
+import { useEffect, useState } from 'react';
+import { useParams, useSearchParams } from 'react-router-dom';
+import { Person } from '../../types';
+import { PersonLink } from '../PerrsonLink/PersonLink';
+import { TableHeadLink } from '../TableHeadLink';
+
+type Props = {
+  filteredPeopleList: Person[];
+};
+
+export const PeopleTable: React.FC<Props> = ({
+  filteredPeopleList,
+}) => {
+  const { urlSlug = '' } = useParams();
+  const [searchParams] = useSearchParams();
+  const sortColumn = searchParams.get('sort') || '';
+  const sortReverse = searchParams.get('order') || '';
+  const [
+    visiblePeopleList,
+    setVisiblePeopleList,
+  ] = useState<Person[]>(filteredPeopleList);
+
+  const isSelected = (slug: string) => slug === urlSlug;
+
+  const parentsCheck = (name: string | null) => {
+    if (!name) {
+      return '-';
+    }
+
+    const linkToParent = visiblePeopleList.find(person => person.name === name);
+
+    return linkToParent ? <PersonLink person={linkToParent} /> : name;
+  };
+
+  enum Persons {
+    name = 'name',
+    sex = 'sex',
+    born = 'born',
+    died = 'died',
+  }
+
+  const tableHeadFilter = async () => {
+    let filteredPeople: Person[] = [...filteredPeopleList].sort((
+      personA: Person,
+      personB: Person,
+    ) => {
+      switch (sortColumn) {
+        case (Persons.name):
+        case (Persons.sex):
+          return personA[sortColumn].localeCompare(personB[sortColumn]);
+
+        case (Persons.born):
+        case (Persons.died):
+          return personA[sortColumn] - personB[sortColumn];
+
+        default:
+          return 0;
+      }
+    });
+
+    if (sortReverse === 'desc') {
+      filteredPeople = filteredPeople.reverse();
+    }
+
+    setVisiblePeopleList(filteredPeople);
+  };
+
+  useEffect(() => {
+    tableHeadFilter();
```

Comment:
The same here. You can do the whole logic inside `useEffect` instead of creating a separate function.